### PR TITLE
Add missing user authentication helper

### PR DESCRIPTION
**Pull request by EnriqueVidal**
This helper methods will make @aalvarado work easier, it will provide him with a callback for controllers he can use as:

``` ruby
before_filter :authenticate_user!
```

Rel: #19

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,9 +1,20 @@
...
     User.find( session[:user_id] ) rescue nil
   end
+
+  private
+  def authenticate_user!
+    raise UserGroup::UserNotAuthenticated unless current_user
+  end
+
+  def unauthenticated_access
+    render :file => 'public/404', :format => :html
```
Comment (javierg) [inline app/controllers/application_controller.rb:18]: I think this should be a 403

Comment (EnriqueVidal) [inline app/controllers/application_controller.rb:18]: That depends, I made it 404 emulating github's behaviour.

Comment (javierg) [inline app/controllers/application_controller.rb:18]: but then when something is missing we will render the same message, btw, I hate that on GH...

**Comment by EnriqueVidal** [comment]
This issue has moved to #19

**Pull request closed without merging**